typespec-tests/src/main/java/com/cadl/armstreamstyleserialization/models/Error.java

```diff
@@ -0,0 +1,190 @@
+        return this.details;
+    }
+
+    /**
+     * Get the additionalProperty property: The additionalProperty property.
+     * 
+     * @return the additionalProperty value.
+     */
+    public String getAdditionalProperty() {
+        return this.additionalProperty;
+    }
+
+    /**
+     * Get the code property: The error code parsed from the body of the http error response.
+     * 
+     * @return the code value.
+     */
+    @Override
+    public String getCode() {
+        return this.code;
+    }
+
+    /**
+     * Get the message property: The error message parsed from the body of the http error response.
+     * 
+     * @return the message value.
+     */
+    @Override
+    public String getMessage() {
+        return this.message;
+    }
+
+    /**
+     * Get the target property: The target of the error.
+     * 
+     * @return the target value.
+     */
+    @Override
+    public String getTarget() {
+        return this.target;
+    }
+
+    /**
+     * Get the additionalInfo property: Additional info for the error.
+     * 
+     * @return the additionalInfo value.
+     */
+    @Override
+    public List<AdditionalInfo> getAdditionalInfo() {
+        return this.additionalInfo;
+    }
+
+    /**
+     * Validates the instance.
+     * 
+     * @throws IllegalArgumentException thrown if the instance is not valid.
+     */
+    public void validate() {
+        if (getDetails() != null) {
+            getDetails().forEach(e -> e.validate());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        return jsonWriter.writeEndObject();
+    }
+
+    /**
+     * Reads an instance of Error from the JsonReader.
+     * 
+     * @param jsonReader The JsonReader being read.
+     * @return An instance of Error if the JsonReader was pointing to an instance of it, or null if it was pointing to
+     * JSON null.
+     * @throws IllegalStateException If the deserialized JSON object was missing any required properties.
+     * @throws IOException If an error occurs while reading the Error.
+     */
+    public static Error fromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            JsonReader bufferedReader = reader.bufferObject();
+            bufferedReader.nextToken();
+            while (bufferedReader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = bufferedReader.getFieldName();
+                bufferedReader.nextToken();
+
+                if ("error".equals(fieldName)) {
+                    return fromJson0(bufferedReader);
+                } else {
+                    bufferedReader.skipChildren();
+                }
+            }
+            return fromJson0(bufferedReader.reset());
+        });
+    }
+
+    private static Error fromJson0(JsonReader reader) throws IOException {
```

Comment:
`fromJson0` :-)

Not picky about private name. core-management is `readManagementError`

Reply:
I used `fromJson0` for generalization. Though `readManagementError` seems good enough too. Updated.